### PR TITLE
Allow to force static linking on Windows in Release

### PR DESCRIPTION
**Pull request by xoofx**
This is related to the issue described on [corefx#14865](https://github.com/dotnet/corefx/issues/14865#issuecomment-270481442)

This PR allows to build a release binaries on Windows without linking to the uCRT by setting the env variable `CLR_CMAKE_WIN32_FORCE_STATIC_LINK` to `true`

### CODE REVIEW

**Comment by danmosemsft** [comment]
@xoofx what is next here? I'm chasing up old PR's. Do you need a CR?

**Comment by jkotas** [comment]
@danmosemsft Feel free to merge this if it looks good. 

I have not merged it myself because of I am oof ... not online enough to deal with issues if there are any.

**Comment by danmosemsft** [comment]
Let's get fresh results. @dotnet-bot test this please

File: CMakeLists.txt [hunk 1]
```diff
@@ -364,11 +364,14 @@ if (WIN32)
     set(CMAKE_EXE_LINKER_FLAGS "${CMAKE_EXE_LINKER_FLAGS} /machine:arm64")
   endif(CLR_CMAKE_PLATFORM_ARCH_ARM64)
 
-  # Force uCRT to be dynamically linked for Release build  
-  set(CMAKE_SHARED_LINKER_FLAGS_RELEASE "${CMAKE_SHARED_LINKER_FLAGS_RELEASE} /NODEFAULTLIB:libucrt.lib /DEFAULTLIB:ucrt.lib")  
-  set(CMAKE_EXE_LINKER_FLAGS_RELEASE "${CMAKE_EXE_LINKER_FLAGS_RELEASE} /NODEFAULTLIB:libucrt.lib /DEFAULTLIB:ucrt.lib")  
-  set(CMAKE_SHARED_LINKER_FLAGS_RELWITHDEBINFO "${CMAKE_SHARED_LINKER_FLAGS_RELWITHDEBINFO} /NODEFAULTLIB:libucrt.lib /DEFAULTLIB:ucrt.lib")  
-  set(CMAKE_EXE_LINKER_FLAGS_RELWITHDEBINFO "${CMAKE_EXE_LINKER_FLAGS_RELWITHDEBINFO} /NODEFAULTLIB:libucrt.lib /DEFAULTLIB:ucrt.lib")  
+  # Force uCRT to be dynamically linked for Release build (unless env variable CLR_CMAKE_WIN32_FORCE_STATIC_LINK is set to true)
+  set(CLR_CMAKE_WIN32_FORCE_STATIC_LINK $ENV{CLR_CMAKE_WIN32_FORCE_STATIC_LINK})
```
Comment (janvorli) [inline CMakeLists.txt]: I don't like using env vars for configuring cmake options. This should be done by adding a new option to the build.cmd that will append extra cmake argument (`-DCLR_CMAKE_WIN32_FORCE_STATIC_LINK=1`).

**Comment by danmosemsft** [comment]
@xoofx are you able to follow this suggestion?

**Comment by danmosemsft** [comment]
@xoofx this is pretty old, I'm going to close it, feel free to reopen if you want to continue, or open a new one

**Pull request closed without merging**